packages/network-controller/tests/provider-api-tests/no-block-param.ts

```diff
@@ -0,0 +1,916 @@
+    await withMockedCommunications({ providerType }, async (comms) => {
+      const request = { method };
+
+      // The first time a block-cacheable request is made, the latest block
+      // number is retrieved through the block tracker first. It doesn't
+      // matter what this is — it's just used as a cache key.
+      comms.mockNextBlockTrackerRequest();
+      // Here we have the request fail for the first 4 tries, then succeed
+      // on the 5th try.
+      comms.mockRpcCall({
+        request,
+        error: 'ETIMEDOUT: Some message',
+        times: 4,
+      });
+
+      comms.mockRpcCall({
+        request,
+        response: {
+          result: 'the result',
+          httpStatus: 200,
+        },
+      });
+
+      const result = await withNetworkClient(
+        { providerType },
+        async ({ makeRpcCall, clock }) => {
+          return await waitForPromiseToBeFulfilledAfterRunningAllTimers(
+            makeRpcCall(request),
+            clock,
+          );
+        },
+      );
+
+      expect(result).toStrictEqual('the result');
+    });
+  });
+
+  // Both the Infura and fetch middleware detect ETIMEDOUT errors and will
+  // automatically retry the request to the RPC endpoint in question, but both
+  // produce a different error if the number of retries is exhausted.
+  if (providerType === 'infura') {
+    it('causes a request to fail with a custom error if an "ETIMEDOUT" error is thrown while making the request to the RPC endpoint 5 times in a row', async () => {
+      await withMockedCommunications({ providerType }, async (comms) => {
+        const errorMessage = 'ETIMEDOUT: Some message';
+        return returnsErrorAfterRetries(
+          providerType,
+          method,
+          errorMessage,
+          buildInfuraClientRetriesExhaustedErrorMessage(errorMessage),
+          comms,
+        );
+      });
+    });
+  } else {
+    it('returns an empty response if an "ETIMEDOUT" error is thrown while making the request to the RPC endpoint 5 times in a row', async () => {
+      await withMockedCommunications({ providerType }, async (comms) => {
+        const request = { method };
+        const errorMessage = 'ETIMEDOUT: Some message';
+
+        // The first time a block-cacheable request is made, the latest block
+        // number is retrieved through the block tracker first. It doesn't
+        // matter what this is — it's just used as a cache key.
+        comms.mockNextBlockTrackerRequest();
+        comms.mockRpcCall({
+          request,
+          error: errorMessage,
+          times: 5,
+        });
+        const result = await withNetworkClient(
+          { providerType },
+          async ({ makeRpcCall, clock }) => {
+            return await waitForPromiseToBeFulfilledAfterRunningAllTimers(
+              makeRpcCall(request),
+              clock,
+            );
+          },
+        );
+
+        expect(result).toBeUndefined();
```

Comment:
This behavior is different from the extension, which returns an "empty response" error.